doc/developer-guides/coding_guidelines.rst

```diff
@@ -2787,3 +2722,418 @@ Compliant example::
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       cpuid(CPUID_EXTEND_FEATURE, &unused, &boot_cpu_data.cpuid_leaves[FEAT_7_0_EBX], &boot_cpu_data.cpuid_leaves[FEAT_7_0_ECX], &boot_cpu_data.cpuid_leaves[FEAT_7_0_EDX]);
+
+
+GN-02: Each line shall only contain one statement
+-------------------------------------------------
+
+Compliant example::
+
+    if (condition) {
+        do_a();
+    } else {
+        do_b();
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       if (condition) { do_a();} else {do_b();}
+
+
+GN-03: The semicolon ending a statement shall be the last character of that line
+--------------------------------------------------------------------------------
+
+Compliant example::
+
+    uint32_t a;
+    uint32_t b;
+    uint32_t c;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t a;uint32_t b;uint32_t c;
+
+
+
+Indentation
+===========
+
+IT-01: Tabs shall be used for code indentation
+----------------------------------------------
+
+Spaces are allowed only for indenting comments or aligning statements that span
+multiple lines.
+
+Compliant example::
+
+    if (condition) {
+        do_a();
+    } else {
+        do_b();
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       if (condition) {
+         do_a();
+       } else {
+         do_b();
+       }
+
+
+IT-02: Tabs shall be 8 characters wide
+--------------------------------------
+
+A tab character shall be considered 8-character wide when limiting the line
+width.
+
+Compliant example::
+
+    /* the following line hits the line limit (120 characters) if tab is counted as 8-character wide */
+            /*1---------2---------3---------4---------5---------6---------7---------8---------9---------A---------B-------*/
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       /* the following line hits the line limit (120 characters) if tab is counted as 4-character wide */
+           /*1---------2---------3---------4---------5---------6---------7---------8---------9---------A---------B-------$$$$*/
+
+
+
+Whitespaces
+===========
+
+WS-01: Trailing whitespaces shall not be allowed at the end of lines
+--------------------------------------------------------------------
+
+Compliant example::
+
+    uint32_t a;
+    uint32_t b;
+    uint32_t c;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t a;    
+       uint32_t b;    
+       uint32_t c;    
+       
+
+
+WS-02: A single space shall exist between non-function-like keywords and opening brackets
+-----------------------------------------------------------------------------------------
+
+A single space shall exist between a non-function-like keyword and the opening
+bracket (either a brace or a parenthesis) that follows. This rule applies to the
+keywords `if`, `else`, `for`, `do`, `while`, `switch` and `return`. This rule
```

Comment:
Because the back tick (`` ` ``) is significant in reST files, use an apostrophe (`` ' ``) or not use one at all.